pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -450,16 +453,44 @@ func (gce *GCECloud) EnsureTCPLoadBalancer(name, region string, loadBalancerIP n
 	return status, nil
 }
 
-// This is kind of hacky, but the managed instance group adds 4 random chars and a hyphen
-// to the base name. Older naming schemes put a hyphen and an incrementing index after
-// the base name. Thus we pull off the characters after the final dash to support both.
-func (gce *GCECloud) computeHostTag(host string) string {
-	host = strings.SplitN(host, ".", 2)[0]
-	lastHyphen := strings.LastIndex(host, "-")
-	if lastHyphen == -1 {
-		return host
+// We grab all tags from all instances being added to the pool.
+// * The longest tag that is a prefix of the instance name is used
+// * If any instance has a prefix tag, all instances must
+// * If no instances have a prefix tag, no tags are used
+func (gce *GCECloud) computeHostTags(hosts []string) ([]string, error) {
+	listCall := gce.service.Instances.List(gce.projectID, gce.zone)
+
+	// Add the filter for hosts
+	listCall = listCall.Filter("name eq (" + strings.Join(hosts, "|") + ")")
+
+	// Add the fields we want
+	listCall = listCall.Fields("items(name,tags)")
+
+	res, err := listCall.Do()
+	if err != nil {
+		return nil, err
 	}
-	return host[:lastHyphen]
+
+	tags := sets.NewString()
+	for _, instance := range res.Items {
+		longest_tag := ""
+		for _, tag := range instance.Tags.Items {
+			if strings.HasPrefix(instance.Name, tag) && len(tag) > len(longest_tag) {
+				longest_tag = tag
+			}
+		}
+		if len(longest_tag) > 0 {
+			tags.Insert(longest_tag)
+		} else if len(tags) > 0 {
+			return nil, fmt.Errorf("Some, but not all, instances have prefix tags (%s is missing)", instance.Name)
+		}
+	}
+
+	if len(tags) == 0 {
+		glog.V(2).Info("No instances had tags, creating rule without target tags")
```

Comment:
See  #14430, creating the rule without a tags should not be an error case if none of the instances have tags.